flink-kubernetes/src/main/java/org/apache/flink/kubernetes/KubernetesResourceManager.java

```diff
@@ -232,57 +229,75 @@ private void recoverWorkerNodesFromPreviousAttempts() throws ResourceManagerExce
 			++currentMaxAttemptId);
 	}
 
-	private void requestKubernetesPod() {
-		numPendingPodRequests++;
+	private void requestKubernetesPod(WorkerResourceSpec workerResourceSpec) {
+		final KubernetesTaskManagerParameters parameters =
+			createKubernetesTaskManagerParameters(workerResourceSpec);
+
+		podWorkerResources.put(parameters.getPodName(), workerResourceSpec);
+		final int pendingWorkerNum = pendingWorkerCounter.increaseAndGet(workerResourceSpec);
 
 		log.info("Requesting new TaskManager pod with <{},{}>. Number pending requests {}.",
-			defaultMemoryMB,
-			defaultCpus,
-			numPendingPodRequests);
+			parameters.getTaskManagerMemoryMB(),
+			parameters.getTaskManagerCPU(),
+			pendingWorkerNum);
+		log.info("TaskManager {} will be started with {}.", parameters.getPodName(), workerResourceSpec);
+
+		final KubernetesPod taskManagerPod =
+			KubernetesTaskManagerFactory.createTaskManagerComponent(parameters);
+		kubeClient.createTaskManagerPod(taskManagerPod);
+	}
+
+	private KubernetesTaskManagerParameters createKubernetesTaskManagerParameters(WorkerResourceSpec workerResourceSpec) {
+		// TODO: need to unset process/flink memory size from configuration if dynamic worker resource is activated
```

Comment:
The purpose of this `TODO` is to avoid conflict between `workerResourceSpec` and `process.size`/`flink.size` in `flinkConfig`. It is not about forbidden workers with different sizes. Actually, one of the main purpose of this PR is to make the RMs not assuming workers have the same size.

I was thinking about change the `Configuration` at creating the `SlotManager`, where we known which implementation of `SlotManager` is used and unset flink/process size if the plugin supports dynamic worker sizes.

Reply:
So do we have the requirement that what's written in the config and what's specified in `workerResourceSpec` have to be the same? If yes, then I think we should add a check state here. This will ensure that we remember what needs to be adjusted in order to support dynamic worker resources.

Reply:
I don't think we have that requirement.

The purpose of unsetting flink/process size from configuration, is to make sure the values in `workerResourceSpec` (task heap, task off-heap, network, managed) and those not in `workerResorceSpec`(framework heap, framework off-heap, jvm metaspace, jvm overhead, total flink, total process) can put together w/o conflict.

If they cannot be put together, we don't really need a check state to remind us because the generating of `taskExecutorProcessSpec` will fail.

Reply:
Ok, instead of creating a TODO, I would suggest to create a JIRA ticket which is linked as a follow up. TODO's tend to be forgotten.

Reply:
I've created FLINK-17061 to track this issue.